### PR TITLE
Add regression test for tasty reflect case study

### PR DESCRIPTION
**Pull request by biboudis**
This PR pushes a toy interpreter for TASTy expressions. The main test file is on the `tasty-interpreter/InterpretedMain.scala` file where we support value declarations, if, while, arithmetic, method definitions, method calls (interpreted or via a reflective proxy if they are precompiled).

Also small additions to `SymbolOps`:
- `isDefinedInCurrentRun`
- `signature` in `DefSymbolDeco`
- `moduleClass` in `ValSymbolDeco`
- `classSymbol` in `TypeDeco`

### CODE REVIEW

**Comment by nicolasstucki** [comment]
@biboudis something failed after the rebase.

File: compiler/src/dotty/tools/dotc/tastyreflect/SymbolOpsImpl.scala [hunk 1]
```diff
@@ -181,8 +189,8 @@ trait SymbolOpsImpl extends scala.tasty.reflect.SymbolOps with CoreImpl {
   def ValSymbolDeco(symbol: ValSymbol): ValSymbolAPI = new ValSymbolAPI {
     def tree(implicit ctx: Context): ValDef = FromSymbol.valDefFromSym(symbol)
 
-    def companionClass(implicit ctx: Context): Option[ClassSymbol] = {
-      val sym = symbol.companionClass
```
Comment (nicolasstucki) [inline compiler/src/dotty/tools/dotc/tastyreflect/SymbolOpsImpl.scala]: Companion class and module class are two different things

Comment (biboudis) [reply compiler/src/dotty/tools/dotc/tastyreflect/SymbolOpsImpl.scala]: Noted. Will fix.

**Comment by biboudis** [comment]
Rebased

File: library/src/scala/tasty/reflect/FlagSet.scala [hunk 2]
```diff
@@ -0,0 +1,88 @@
+package scala.tasty.reflect
+
+trait FlagSet {
```
[the remark this answers is not in the document]
Comment (biboudis) [reply library/src/scala/tasty/reflect/FlagSet.scala]: And I was trying to track it down yesterday because it wasn't used anywhere. 😛

**Comment by biboudis** [comment]
@dotty-bot recheck cla

**Comment by biboudis** [comment]
@dotty-bot recheck cla

**Pull request merged**